src/main/java/de/epiceric/shopchest/utils/PluginMessageTask.java

```diff
@@ -0,0 +1,34 @@
+package de.epiceric.shopchest.utils;
+
+import de.epiceric.shopchest.ShopChest;
+import org.bukkit.entity.Player;
+import org.bukkit.scheduler.BukkitRunnable;
+
+import java.io.ByteArrayOutputStream;
+
+public class PluginMessageTask extends BukkitRunnable {
```

Comment:
Is this class really needed? I feel like you could have a simple 
```java
plugin.getScheduler().runTaskAsynchronously(() -> player.sendPluginMessage(plugin, "BungeeCord", message));
```
in the `sendBungeeMessage()` method and use the player who bought/sold as the online player.

Reply:
https://github.com/xPiston/ShopChest/commit/8ec8a63e4b8151de08dd18ba1ec1b7c140b0f0aa

Reply:
I mean, you could remove the whole `PluginMessageTask` class and use this as the `sendBungeeMessage()`:
```java
public void sendBungeeMessage(String player, String message) {
    try {
        ByteArrayOutputStream b = new ByteArrayOutputStream();
        DataOutputStream out = new DataOutputStream(b);

        out.writeUTF("Message");
        out.writeUTF(player);
        out.writeUTF(message);

        if (!plugin.getServer().getOnlinePlayers().isEmpty()) {
            plugin.getServer().getScheduler().runTaskAsynchronously(plugin, () -> {
                Player player = plugin.getServer().getOnlinePlayers().iterator().next();
                player.sendPluginMessage(plugin, "BungeeCord", b.toByteArray());
            });
        }
    } catch (Exception e) {
        plugin.debug("Failed to send bungee message");
        plugin.debug(e);
        plugin.getLogger().warn("Failed to send BungeeCord message")
    }
}
```